patches/chrome-browser-resources-tools-rollup_plugin.js.patch

```diff
@@ -0,0 +1,59 @@
+diff --git a/chrome/browser/resources/tools/rollup_plugin.js b/chrome/browser/resources/tools/rollup_plugin.js
+index e1942718e76890c5051923fc647f488c1801ca95..c276588f6086137b6022d694fb00719f1dc9decc 100644
+--- a/chrome/browser/resources/tools/rollup_plugin.js
++++ b/chrome/browser/resources/tools/rollup_plugin.js
+@@ -72,13 +72,14 @@ export default function plugin(srcPath, genPath, rootPath, host, excludes) {
+       }
+ 
+       // Get path from ui/webui/resources
++      const braveResourcesUrl = 'chrome://brave-resources/'
```

Comment:
this patch is a major problem, please dm to discuss

Reply:
@bsclifton 2 ways I can see to clean up:

1) Move the "Brave block" to _before_ the "Chromium block", then we can remove the `source.indexOf...` modifications to the chromium lines. We need to do the same `source.indexOf...` modifications to the Brave lines (but for the Chromium url). Then replace that with a function call to a .js file somewhere in `brave/browser/resources/...`. This would result in a 2-line patch (1 for the import and 1 for the function call).

2) Don't patch this file at all. Instead add a new `brave_rollup_plugin.js` file, and get `optimize_webui.py` to add our plugin JS to the rollup config. We'd do this by adding something to `_generate_rollup_config` to `import bravePlugin from '[...bravepath]`. Perhaps a nice way to do this and cleanup the patch for `optimize_webui.py` too would be to create a function in a brave python module called `rollup_extra_js_config()` which would return some extra path(s) to add to plugins in the file. Still requires 2 lines of patching though since it needs to be imported and called 😞 . I do have [a cleanup for `opimize_webui` in an unmerged PR here](https://github.com/brave/brave-core/pull/2911/files#diff-03ea122b9c28c8c3e0ff9f0c1ced9d52R8) which we could use if we're adding any more patches to this file.

Reply:
Because the `rollup_plugin.js` is being passed into a Node.js context (executed in Python), patching in a `require()` is definitely messy (path is relative to the node module). I did make the patch cleaner, but I think going with an option 2 is the way to go. Will look into this tomorrow